### DISCUSSION TITLE
Integrating External Sources

### DISCUSSION THREAD

**Discussion by sascharau** [mysticaltech/terraform-hcloud-kube-hetzner#1061] [Q&A]
Hi,
i currently have cx11 instances under very old conditions (2 EUR). Is it possible to integrate these without causing any issues with Terraform?

My strategy involved creating cx11 instances through Terraform, taking snapshots, and then deploying these snapshots onto the old instances. However, this approach failed at the instance creation step. Here's the error message:

module.kube-hetzner.module.agents["5-0-agent-xsmall"].data.cloudinit_config.config: Read complete after 0s [id=80283578]
module.kube-hetzner.module.agents["5-0-agent-xsmall"].hcloud_server.server: Creating...
```
╷
│ Error: image disk is bigger than server type disk (service_error)
│ 
│   with module.kube-hetzner.module.agents["5-0-agent-xsmall"].hcloud_server.server,
│   on .terraform/modules/kube-hetzner/modules/host/main.tf line 22, in resource "hcloud_server" "server":
│   22: resource "hcloud_server" "server" {
│ 
╵
```

cx11 has 20 GB, but the image has only 0.6 GB.

Can anyone assist me?

**Comment by Silvest89** [comment]
You cannot do snapshots, since a snapshot is a copy of a server disk. Your old legacy CX11 instances probably have much less total disk space than the newer ones

**Comment by sascharau** [comment]
Thank you for your response! 
I may not have been clear in my previous message. Is it possible that the CX11 option (1VCPU, 2GB RAM, 20GB SSD) is not available? I receive the aforementioned error when I run the following settings:

```
agent_nodepools = [
  {
    name        = "agent-xsmall",
    server_type = "cx11", #not cpx11
    location    = "fsn1",
    labels      = [],
    taints      = [],
    count       = 1

  },
]
```

**Reply by Silvest89** [reply]
```
For the server type, the minimum instance supported is cpx11 (just a few cents more than cx11) but the cax11 is even cheaper and more powerful (arm64); see https://www.hetzner.com/cloud. It states so in the kube.tf.example
```
I haven't tried a CX11, but I wouldn't want to host my cluster on those nodes anyway

**Comment by sascharau** [comment]
Apparently the hard drive is too small.  Even with the Hetzner UI I cannot import a snapshot with MicroOS.

Thanks for your time.
